packages/expo-updates/ios/EXUpdates/EXUpdatesDevLauncherController.m

```diff
@@ -0,0 +1,119 @@
+//  Copyright © 2021 650 Industries. All rights reserved.
+
+#import <EXUpdates/EXUpdatesAppController+Internal.h>
+#import <EXUpdates/EXUpdatesAppLauncherWithDatabase.h>
+#import <EXUpdates/EXUpdatesConfig.h>
+#import <EXUpdates/EXUpdatesDevLauncherController.h>
+#import <EXUpdates/EXUpdatesReaper.h>
+#import <EXUpdates/EXUpdatesReaperSelectionPolicyDevelopmentClient.h>
+#import <EXUpdates/EXUpdatesRemoteAppLoader.h>
+#import <EXUpdates/EXUpdatesSelectionPolicy.h>
+#import <EXUpdates/EXUpdatesUpdate.h>
+#import <React/RCTBridge.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+static NSString * const EXUpdatesDevLauncherControllerErrorDomain = @"EXUpdatesDevLauncherController";
+
+@implementation EXUpdatesDevLauncherController
+
+@synthesize bridge = _bridge;
+
++ (instancetype)sharedInstance
+{
+  static EXUpdatesDevLauncherController *theController;
+  static dispatch_once_t once;
+  dispatch_once(&once, ^{
+    if (!theController) {
+      theController = [EXUpdatesDevLauncherController new];
+    }
+  });
+  return theController;
+}
+
+- (void)setBridge:(nullable id)bridge
+{
+  _bridge = bridge;
+  if ([bridge isKindOfClass:[RCTBridge class]]) {
+    EXUpdatesAppController.sharedInstance.bridge = (RCTBridge *)bridge;
+  }
+}
+
+- (NSURL *)launchAssetURL
+{
+  return EXUpdatesAppController.sharedInstance.launchAssetUrl;
+}
+
+- (void)fetchUpdateWithConfiguration:(NSDictionary *)configuration
+                          onManifest:(EXUpdatesManifestBlock)manifestBlock
+                            progress:(EXUpdatesProgressBlock)progressBlock
+                             success:(EXUpdatesSuccessBlock)successBlock
+                               error:(EXUpdatesErrorBlock)errorBlock
+{
+  EXUpdatesAppController *controller = EXUpdatesAppController.sharedInstance;
+  EXUpdatesConfig *updatesConfiguration = [EXUpdatesConfig configWithExpoPlist];
+  [updatesConfiguration loadConfigFromDictionary:configuration];
+  if (!updatesConfiguration.updateUrl) {
+    errorBlock([NSError errorWithDomain:EXUpdatesDevLauncherControllerErrorDomain code:1 userInfo:@{NSLocalizedDescriptionKey: @"Failed to load update: configuration object must include a valid update URL"}]);
```

Comment:
suggestion - define an enum for error codes to make it easier to keep them stable over time:

```
typedef NS_ENUM(NSInteger, EXUpdatesDevLauncherErrorCode) {
    EXUpdatesDevLauncherErrorCodeInvalidUpdateURL = 1,
    EXUpdatesDevLauncherErrorCodeDirectoryInitializationFailed,
    EXUpdatesDevLauncherErrorCodeDatabaseInitializationFailed,
    EXUpdatesDevLauncherErrorCodeUpdateLaunchFailed,
};
``